### PR TITLE
Add prefix stop

### PR DESCRIPTION
**Pull request by wellsjo**
cc @jwerle

### CODE REVIEW

File: lib/Request.js [hunk 1]
```diff
@@ -240,19 +232,17 @@ class Request {
...
           batch.push(done => {
 
-            let b = source.bucket;
-            let k = source.key;
-            let e = this.encoding;
-            let t = this.transformer;
+            let bucket = source.bucket;
+            let key = source.key;
 
-            this.s3.get(b, k, e, t).then(val => {
```
Comment (jwerle) [inline lib/Request.js]: lol

Comment (jwerle) [inline lib/Request.js]: almost spells out bucket

File: lib/S3.js [hunk 2]
```diff
@@ -287,78 +286,42 @@ class S3 {
...
+            let stopPrefixReached = false
 
-    let self = this;
-    let allKeys = [];
-    marker = marker || '';
+            if (endPrefix.length > 0) {
+              for (let i = 0; i < keyValues.length; i++) {
+                if (keyValues[i].indexOf(endPrefix) > -1) {
+                  keyValues = keyValues.slice(0, i)
+                  stopPrefixReached = true
```
Comment (jwerle) [inline lib/S3.js]: is `stopPrefixReached` ever used?

File: lib/Request.js (hunk 1 above)
Comment (wellsjo) [inline lib/Request.js]: lol

File: lib/S3.js (hunk 2 above)
Comment (wellsjo) [inline lib/S3.js]: I should prob run this through the linter..lol

Comment (wellsjo) [inline lib/S3.js]: yeah it's used line `300` and `310`

**Pull request merged**